verify-bazel failing with 'cannot find package "github.com/bazelbuild/buildtools/build"' etc

/kind bug

verify-bazel is currently failing with the following stacktrace:
```
$ hack/verify-bazel.sh 
../../internal/config/directives.go:24:2: cannot find package "github.com/bazelbuild/buildtools/build" in any of:
        /tmp/tmp.RKacdj87fR/go/src/github.com/bazelbuild/bazel-gazelle/vendor/github.com/bazelbuild/buildtools/build (vendor tree)
        [path]/.gvm/gos/go1.9.4/src/github.com/bazelbuild/buildtools/build (from $GOROOT)
        /tmp/tmp.RKacdj87fR/go/src/github.com/bazelbuild/buildtools/build (from $GOPATH)
../../internal/rules/construct.go:27:2: cannot find package "github.com/bazelbuild/buildtools/tables" in any of:
        /tmp/tmp.RKacdj87fR/go/src/github.com/bazelbuild/bazel-gazelle/vendor/github.com/bazelbuild/buildtools/tables (vendor tree)
        [path]/.gvm/gos/go1.9.4/src/github.com/bazelbuild/buildtools/tables (from $GOROOT)
        /tmp/tmp.RKacdj87fR/go/src/github.com/bazelbuild/buildtools/tables (from $GOPATH)
!!! [0302 11:47:54] Call tree:
!!! [0302 11:47:54]  1: ./hack/update-bazel.sh:34 kube::util::go_install_from_commit(...)
!!! Error in /tmp/verify-bazel.5wHzRb/go/src/k8s.io/kubernetes/hack/lib/util.sh:483
  Error in /tmp/verify-bazel.5wHzRb/go/src/k8s.io/kubernetes/hack/lib/util.sh:483. '((i<3-1))' exited with status 1
Call stack:
  1: /tmp/verify-bazel.5wHzRb/go/src/k8s.io/kubernetes/hack/lib/util.sh:483 kube::util::go_install_from_commit(...)
  2: ./hack/update-bazel.sh:34 main(...)
Exiting with status 1
!!! [0302 11:47:54] Call tree:
!!! [0302 11:47:54]  1: ./hack/update-bazel.sh:34 kube::util::go_install_from_commit(...)
!!! Error in /tmp/verify-bazel.5wHzRb/go/src/k8s.io/kubernetes/hack/lib/util.sh:473
  Error in /tmp/verify-bazel.5wHzRb/go/src/k8s.io/kubernetes/hack/lib/util.sh:473. '((i<3-1))' exited with status 1
Call stack:
  1: /tmp/verify-bazel.5wHzRb/go/src/k8s.io/kubernetes/hack/lib/util.sh:473 kube::util::go_install_from_commit(...)
  2: ./hack/update-bazel.sh:34 main(...)
Exiting with status 1
!!! Error in hack/verify-bazel.sh:43
  Error in hack/verify-bazel.sh:43. 'GOPATH="${_tmp_gopath}" ./hack/update-bazel.sh' exited with status 1
Call stack:
  1: hack/verify-bazel.sh:43 main(...)
Exiting with status 1
```

Ironically, this is caused by a fix upstream which would probably prevent #60442 from occurring again - gazelle vendored `github.com/bazelbuild/buildtools` in https://github.com/bazelbuild/bazel-gazelle/pull/145, which means that when we hackily download gazelle's dependencies with `go get -d`, we no longer also download the `github.com/bazelbuild/buildtools` package.

We then checkout an older version of gazelle, which hasn't vendored `github.com/bazelbuild/buildtools`, and so the dependency is missing, and we fail to build.

For the master branch, we can probably just bump to latest gazelle. The older release branches will probably require more hacks.

Happy Friday!
/assign

Not surprisingly I am seeing this issue when I run hack/update-bazel.sh:

./../internal/config/directives.go:24:2: cannot find package "github.com/bazelbuild/buildtools/build" in any of:
...

Is there any workaround?

--- a/hack/update-bazel.sh
+++ b/hack/update-bazel.sh
@@ -32,7 +32,7 @@ kube::util::go_install_from_commit \
     97099dccc8807e9159dc28f374a8f0602cab07e1
 kube::util::go_install_from_commit \
     github.com/bazelbuild/bazel-gazelle/cmd/gazelle \
-    a85b63b06c2e0c75931e57c4a1a18d4e566bb6f4
+    578e73e57d6a4054ef933db1553405c9284322c7
 
 touch "${KUBE_ROOT}/vendor/BUILD"
 

